onnxruntime/core/graph/graph.cc

```diff
@@ -417,10 +411,10 @@ common::Status NodeArg::UpdateTypeAndShape(const ONNX_NAMESPACE::TypeProto& inpu
 #endif
 
     case TypeProto::kOptionalType: {
-      if (utils::HasOptionalTensorType(input_type) && !utils::HasOptionalTensorType(current_type)) {
-        return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Optional Type mismatch. Expected: Seq tensor. Got: Tensor");
-      } else if (!utils::HasOptionalTensorType(input_type) && utils::HasOptionalTensorType(current_type)) {
-        return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Optional Type mismatch. Expected: Tensor. Got: Seq Tensor");
+      if ((utils::HasOptionalTensorType(input_type) && !utils::HasOptionalTensorType(current_type)) ||
+          (!utils::HasOptionalTensorType(input_type) && utils::HasOptionalTensorType(current_type))) {
+        return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Optional Type mismatch. Expected: ", current_type,
```

Comment:
Is this more like INVALID_ARGUMENT?

Reply:
Typically all type merge mis-matches are treated as FAIL (see above). I do not have a strong preference and just kept it consistent.

Reply:
as it's not a user provided value would 'invalid argument' be confusing?